tests/parsing/GlobbingTest.py

```diff
@@ -218,8 +218,8 @@ def setUp(self):
     def _test_glob(self, pattern, file_list):
         results = sorted([os.path.normcase(g) for g in glob(pattern)])
         file_list = sorted([os.path.normcase(f) for f in file_list])
-        self.assertEqual([i for i in results if "__pycache__" not in i],
-                         file_list)
+        self.assertEqual([i for i in results if not any(x in i
+                          for x in ("__pycache__", ".pyc"))], file_list)
```

Comment:
Your code seems quite complex ...
Can't you use regex ? Would make it simple
... If re.search(r"(__pycache__|\.pyc)", I) ...

Also, try indenting it differently so that it doesn't cause max line length error